src/prefect/utilities/visualization.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any
+import graphviz
+
+
+class FlowVisualizationError(Exception):
+    pass
+
+
+class TaskRunTrackerState:
+    current = None
+
+
+def get_task_run_tracker():
+    return TaskRunTrackerState.current
+
+
+def track_task_run(task_name, parameters, mock_return=None):
+    task_run_tracker = get_task_run_tracker()
+    if task_run_tracker:
+        upstream_tasks = []
+        for k, v in parameters.items():
```

Comment:
how deep do we look for task runs? (this is depth 1) what does the UI support? 
maybe need to use `visit_collection`?